### PR TITLE
Implemented some extra checks and cleaned up a bit

### PR DESCRIPTION
**Pull request by Darjusch**
You should not be able to register a account with: "acceptedTerms": false,
    "hasRequiredAge": false
 
You should not be able to create the product with the same barcode multiple times.
	
You should not be able to create the product with the field verified.

We don’t need to show the product id and the state.

We should return only verified products

Get product should only return verified product

The get product validator is checking the wrong thing its not checking the parameter

The get product should not show the _id and the _v 

Only once account with the same username 

swagger rerouting at /api

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -3,10 +3,10 @@ import mongoose from 'mongoose';
 import bodyParser from 'body-parser';
 import morgan from 'morgan';
 import expressValidator from 'express-validator';
-import cors from 'cors';
```
Comment (thatsddr) [inline src/app.ts:6]: Why has CORS been removed?

File: src/controllers/auth.ts [hunk 2]
```diff
@@ -12,16 +12,21 @@ const registerCounter = new client.Counter({
 
 export const registerUser = (req: Request, res: Response) => {
 
-    // we have to send a email that bestätigt that it is ur email
     // send better responses
```
Comment (thatsddr) [inline src/controllers/auth.ts:15]: Should we add more informative responses on the front end as well?

**Review by thatsddr — COMMENTED** [review]
The code looks good but I think I have an old API key in my environmental variables and I wasn't able to successfully run the tests because of that

File: src/controllers/auth.ts (hunk 2 above)
Comment (5h3rr1ll) [reply src/controllers/auth.ts:15]: ".. that 'bestätigt' ..." 😂

@thatsddr actually it depends a bit what kind of response @Darjusch means here. Response codes or response messages, second we need to be careful with.

**Comment by 5h3rr1ll** [comment]
@Darjusch the Travis CI tests are failing, did u check why?

File: src/controllers/auth.ts (hunk 2 above)
Comment (thatsddr) [reply src/controllers/auth.ts:15]: @5h3rr1ll a couple of lines after that comment there are messages like: "User does not exist or password/email is wrong"

File: src/app.ts (hunk 1 above)
Comment (Darjusch) [reply src/app.ts:6]: because it didn't work in combination with daves refactoring :D

File: src/controllers/auth.ts (hunk 2 above)
Comment (Darjusch) [reply src/controllers/auth.ts:15]: yeah i mean mostly messages but some response codes could also be thought about. 
I think its more for the backend to not give away to much information. 
The example u gave domenico is fine.

**Review by 5h3rr1ll — APPROVED** [review]
Should work now 👍

**Pull request merged**